docs/Guides/reading-mcap-files/json-2.md

```diff
@@ -0,0 +1,184 @@
+
+```python
+float32 = 7  # as defined in the schema
+pointcloud = {
+  "point_stride": (4 + 4 + 4 + 4),  # four bytes per float
+  "fields": [
+    {"name": "x", "offset": 0, "type": float32},
+    {"name": "y", "offset": 4, "type": float32},
+    {"name": "z", "offset": 8, "type": float32},
+    {"name": "i", "offset": 12, "type": float32},
+  ],
+}
+```
+
+Next, let’s pack the points using Python’s built-in [`struct`](https://docs.python.org/3/library/struct.html) and [`base64`](https://docs.python.org/3/library/base64.html) libraries.
+
+```python
+points = bytearray()
+base_timestamp = None
+for point_timestamp, intensity, x, y, z in point_reader(args.csv):
+  if base_timestamp is None:
+    base_timestamp = point_timestamp
+  points.extend(struct.pack("&lt;ffff", x, y, z, intensity))
+assert base_timestamp is not None, "found no points in input csv"
+pointcloud["data"] = base64.b64encode(points).decode("utf-8")
+```
+
+Each 3D object must exist in its own coordinate frame. A point cloud’s `frame_id` identifies the coordinate frame it belongs in, and its `pose` determines its relative position from that coordinate frame’s center. Since we will only have one coordinate frame in our MCAP file, we can choose any arbitrary string as our `frame_id`, and use the identity pose to place our point cloud in its center.
+
+```python
+pointcloud["pose"] = {
+  "position": {"x": 0, "y": 0, "z": 0},
+  "orientation": {"x": 0, "y": 0, "z": 0, "w": 1},
+}
+pointcloud["frame_id"] = "lidar"
+```
+
+We’ll leave the timestamp field for later, when we write the messages into the MCAP file.
+
+### Creating a writer
+
+We’ll start with some imports from the Python MCAP library:
+
+```python
+from mcap.mcap0.writer import Writer
+from mcap.mcap0.well_known import SchemaEncoding, MessageEncoding
+```
+
+Open a file where we’ll output our MCAP data and write our header:
+
+```python
+with open(args.output, "wb") as f:
+  writer = Writer(f)
+  writer.start("x-jsonschema", library="my-excellent-library") # can use any name, as we are not using an MCAP profile: https://www.google.com/url?q=https://github.com/foxglove/mcap/blob/main/docs/specification/appendix.md%23well-known-profiles&sa=D&source=docs&ust=1658261499762935&usg=AOvVaw2GgwRzJbSlqWOErSFru7L_
+```
+
+### Register channel
+
+Let's create a channel of messages to contain our point cloud. The schema's name and content tell Foxglove Studio that it can parse and display this message as a point cloud.
+
+```python
+with open(Path(__file__).parent / "PointCloud.json", "rb") as f:
+  schema = f.read()
+schema_id = writer.register_schema(
+  name="foxglove.PointCloud",
+  encoding=SchemaEncoding.JSONSchema,
+  data=schema,
+)
+channel_id = writer.register_channel(
+  topic="pointcloud",
+  message_encoding=MessageEncoding.JSON,
+  schema_id=schema_id,
+)
+```
+
+### Write messages
+
+Let's write a single `foxglove.PointCloud` message on the channel we just created:
```

Comment:
this file has a mix of characters for single quote/apostrophe. Here it's `Let's` while a bit higher up it's `We’ll`. Not sure what is causing the difference (different editors maybe?) but the first one (`Let's`) seems more standard to me.